src/Composer/Command/BaseCommand.php

```diff
@@ -153,6 +175,14 @@ protected function initialize(InputInterface $input, OutputInterface $output)
             $input->setOption('no-progress', true);
         }
 
+        if (true == $input->hasOption('no-dev')) {
+            if (true == $input->getOption('no-dev') || true == getenv('COMPOSER_NO_DEV')) {
+                $this->setOptionNoDev(true);
+            } else {
+                $this->setOptionNoDev(false);
+            }
+        }
```

Comment:
IMO it'd be a better/less intrusive change if it simply would set a default value for the option if the option is not provided e.g.

```suggestion
        if (true == $input->hasOption('no-dev') && !$input->getOption('no-dev') && Platform::getEnv('COMPOSER_NO_DEV')) {
            $input->setOption('no-dev', true);
        }
```

Then all other changes in every command can be undone.

Reply:
I kept the two `if`s separate because I thought it was easier to understand that way. Is that an issue?

Can you also clarify the condition? I think it would be `($input->getOption('no-dev') || Platform::getEnv('COMPOSER_NO_DEV'))`.

And, does this work if #10188 hasn't been merged yet?

Reply:
https://github.com/composer/composer/pull/10218 will be merged first if needed, that's no problem.